src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -243,20 +243,28 @@ where supportedLanguages.Any()
 
         private static IEnumerable<string> GetSupportedLanguages(TypeDefinition typeDef, PEModule peModule, Type attributeType, AttributeLanguagesFunc languagesFunc)
         {
+            IEnumerable<string>? result = null;
             foreach (CustomAttributeHandle customAttrHandle in typeDef.GetCustomAttributes())
             {
                 if (peModule.IsTargetAttribute(customAttrHandle, attributeType.Namespace!, attributeType.Name, ctor: out _))
                 {
-                    IEnumerable<string>? attributeSupportedLanguages = languagesFunc(peModule, customAttrHandle);
-                    if (attributeSupportedLanguages != null)
+                    if (languagesFunc(peModule, customAttrHandle) is { } attributeSupportedLanguages)
```

Comment:
should you put some sort of comment saying this is a specialized hotpath and not to change it to be simpler?

Reply:
I thought about it, but it didn't seem like a case that was likely to change in the future. The method is only used in one location, is eagerly evaluated, and the old code had essentially the same complexity.